src/Features/Core/Portable/EmbeddedLanguages/Json/LanguageServices/JsonLanguageDetector.cs

```diff
@@ -15,199 +14,188 @@
 using Microsoft.CodeAnalysis.Shared.Extensions;
 using Roslyn.Utilities;
 
-namespace Microsoft.CodeAnalysis.Features.EmbeddedLanguages.Json.LanguageServices
+namespace Microsoft.CodeAnalysis.Features.EmbeddedLanguages.Json.LanguageServices;
+
+/// <summary>
+/// Helper class to detect json in string tokens in a document efficiently.
+/// </summary>
+internal sealed class JsonLanguageDetector(
+    EmbeddedLanguageInfo info,
+    ISet<INamedTypeSymbol> typesOfInterest)
+    : AbstractLanguageDetector<JsonOptions, JsonTree, JsonLanguageDetector>(info, LanguageIdentifiers, CommentDetector)
 {
-    /// <summary>
-    /// Helper class to detect json in string tokens in a document efficiently.
-    /// </summary>
-    internal class JsonLanguageDetector(
-        EmbeddedLanguageInfo info,
-        ISet<INamedTypeSymbol> typesOfInterest) : AbstractLanguageDetector<JsonOptions, JsonTree>(info, LanguageIdentifiers)
+    public static readonly ImmutableArray<string> LanguageIdentifiers = ImmutableArray.Create("Json");
+    public static readonly EmbeddedLanguageCommentDetector CommentDetector = new(LanguageIdentifiers);
```

Comment:
>EmbeddedLanguageCommentDetector [](http://example.com/codeflow?start=27&length=31)

When someone creates a 5th one of these, will it be obvious enough they should be statically caching it? 

One somewhat ugly item to make that more obvious would be to have EmbeddedLanguageCommentDetector ctor private and instead let it cache a static instance for each of the language identifier groupings in use.

Reply:
enh :)  my caring level is low.  presumably we'll copy the existing pattern that we're using for the other 3.

Reply:
Good enough for me :)